asapdiscovery-alchemy/asapdiscovery/alchemy/schema/network.py

```diff
@@ -184,15 +212,37 @@ def generate_network(
             raise RuntimeError(
                 "The radial type network requires a ligand to act as the central node."
             )
+        if (
+            self.network_planning_method.type == "CustomNetworkPlanner"
+            and not custom_network_file
+        ):
+            raise ValueError(
+                "When setting network_planning_method:type: CustomNetworkPlanner in your free energy perturbation factory, you must provide a custom network CSV (-cn)."
+            )
 
         # build the network planner
-        planner_data = {
-            "ligands": [
-                mol.to_openfe() for mol in ligands
-            ],  # need to convert to rdkit objects?
-            "mappers": [self.atom_mapping_engine.get_mapper()],
-            "scorer": self._get_scorer(),
-        }
+        if custom_network_file:
+            if not self.network_planning_method.type == "CustomNetworkPlanner":
+                raise ValueError(
+                    "When providing a custom network CSV (-cn), you must set network_planning_method:type: CustomNetworkPlanner in your free energy perturbation factory."
+                )
+            planner_data = {  # for reason OpenFE's `generate_network_from_names` takes in `mapper` instead of `mappers`
+                "ligands": [
+                    mol.to_openfe() for mol in ligands
+                ],  # need to convert to rdkit objects?
+                "mapper": self.atom_mapping_engine.get_mapper(),
+                "names": CustomNetworkPlanner.read_custom_network_csv(
+                    custom_network_file
+                ),
```

Comment:
Needs scoring or no?

Reply:
nope, see https://github.com/OpenFreeEnergy/openfe/blob/main/openfe/setup/ligand_network_planning.py#L322-L326. No need for edge-scoring because we're manually defining the edges.